distribution/bwc/build.gradle

```diff
@@ -197,6 +199,32 @@ bwcVersions.forPreviousUnreleased { VersionCollection.UnreleasedVersionInfo unre
         }
     }
 
+    task resolveAllBwcDependencies(type: Exec) {
+        dependsOn checkoutBwcBranch, writeBuildMetadata
+        workingDir = checkoutDir
```

Comment:
I haven't spent too much time trying to do that because I'm thinking about implementing tasks for most of what's here in Java. 
Are you ok with addressing this in a subsequent PR to be able to get benefits in CI faster ?

Reply:
I think the closure would just be a helper method in java calling the appropriate setters, and it doesn't really seem that different from the closure you have already separated out in this PR? I'm concerned there are already a lot of other followups that haven't had priority, and this seems like a simple change to this PR.